clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -476,14 +480,34 @@ public <T> List<T> getConfiguredInstances(List<String> classNames, Class<T> t, M
             return objects;
         Map<String, Object> configPairs = originals();
         configPairs.putAll(configOverrides);
-        for (Object klass : classNames) {
-            Object o = getConfiguredInstance(klass, t, configPairs);
-            objects.add(t.cast(o));
+
+        try {
+            for (Object klass : classNames) {
+                Object o = getConfiguredInstance(klass, t, configPairs);
+                objects.add(t.cast(o));
+            }
+        } catch (Exception e) {
+            for (Object object : objects) {
+                if (object instanceof AutoCloseable) {
+                    try {
+                        ((AutoCloseable) object).close();
+                    } catch (Exception ex) {
+                        log.error(String.format("Close exception on %s", object.getClass().getName()), ex);
+                    }
+                } else if (object instanceof Closeable) {
+                    try {
+                        ((Closeable) object).close();
+                    } catch (Exception ex) {
+                        log.error(String.format("Close exception on %s", object.getClass().getName()), ex);
+                    }
```

Comment:
This can be simplified a bit:

```suggestion
                    Utils.closeQuietly((AutoCloseable) object, "AutoCloseable object constructed and configured during failed call to getConfiguredInstances");

```

1. `Utils::closeQuietly` handles failures for us
2. `Closeable` is a subinterface of `AutoCloseable`, so we only need to check for the latter

Reply:
@C0urante  I'm glad to hear that you are feeling better.  
Yes, given the KIP avoidance constraint, it seemed the only way to address this was to change the getConfiguredInstances.   But,  I believe I've implemented each of your comments and thanks for the excellent feedback.